enhancements/network/service-egress-traffic-steering.md

```diff
@@ -0,0 +1,286 @@
+```yaml
+apiVersion: v1
+kind: Service
+metadata:
+  name: example-service
+  namespace: some-namespace
+  annotations:
+    metallb.universe.tf/address-pool: example-pool
+    k8s.ovn.org/egress-service: '{"network-hint":{"cidr":"172.19.0.0/24"}}'
+spec:
+  selector:
+    app: example
+  ports:
+    - name: http
+      protocol: TCP
+      port: 8080
+      targetPort: 8080
+  type: LoadBalancer
+```
+
+3. Advertise the service from the node in charge of the service's traffic. So far the service is "broken" - it is not reachable from outside the cluster and if the pods try to send traffic outside it would probably not come back as it is SNATed to an IP which is unknown.
+We create the advertisements targeting only the node that is in charge of the service's traffic using the `nodeSelectors` field, relying on ovn-k to label the node properly.
+```yaml
+apiVersion: metallb.io/v1beta1
+kind: L2Advertisement
+metadata:
+  name: example-l2-adv
+  namespace: metallb-system
+spec:
+  ipAddressPools:
+  - example-pool
+  nodeSelectors:
+  - matchLabels:
+      egress-service.k8s.ovn.org/some-namespace-example-service: ""
+---
+apiVersion: metallb.io/v1beta1
+kind: BGPAdvertisement
+metadata:
+  name: example-l2-adv
+  namespace: metallb-system
+spec:
+  ipAddressPools:
+  - example-pool
+  nodeSelectors:
+  - matchLabels:
+      egress-service.k8s.ovn.org/some-namespace-example-service: ""
+```
+While possible to create more advertisements resources for the `example-pool`, it is the user's responsibility to make sure that the pool is advertised only by advertisements targeting the node holding the `egress-service.k8s.ovn.org/<svc-namespace>-<svc-name>: ""` label - otherwise the traffic of the service will be broken.
+### User Stories
+As a user of OpenShift, I should be able to have functioning LoadBalancer services whose backing pods exit the cluster with the service's IP from an interface that belongs to a network I specify.
+#### Story 1
+
+As a Telco customer who uses SNMP in OpenShift, I want to access pods that I'm managing using a LoadBalancer service. In order to do so, I need these pods to send traps using the same IP as I use for polling them.
+
+### API Extensions
+N/A
+
+### Test Plan
+
+- Unit tests coverage.
+- E2E coverage by creating a LoadBalancer service with the proper annotations and validating that:
+  - pod to pod traffic works.
+  - external client to service works.
+  - pod to external client works and is SNATed properly.
+
+
+### Risks and Mitigations
+
+- The solution might be a bit fragile as it relies on the user to configure the external advertisement of the service manually, with certain limitations.
+
+- Using a single node to handle all of the traffic of a given service might be a bottleneck, and we will also need to try electing nodes in a way that spreads handling these kind of services between them. Failover must also be taken care of in case a node handling a service falls.
```

Comment:
Will it be possible to add a section on how we are going to handle the distribution of the ip addresses?
It's fine if we do it in a following PR but lets have this documented as a TODO